src/common/plugin.c

```diff
@@ -1318,11 +1328,41 @@ hexchat_list_free (hexchat_plugin *ph, hexchat_list *xlist)
 int
 hexchat_list_next (hexchat_plugin *ph, hexchat_list *xlist)
 {
-	if (xlist->next == NULL)
-		return 0;
+	int ok = 0; /* this avoids using goto */
+	while (ok^=1) {
```

Comment:
why `^=`

Reply:
starts with ok at 0, turns ok into 1 so the loop runs, then when the loop's gonna run the second time it becomes 0 again and continues normally.

this was literally just to avoid using goto, so it uses a loop instead. this is comparable to (but different than) setting it to 1 at the start of the loop, but only uses one line.